example/src/PaymentMethodsView.js

```diff
@@ -18,43 +18,71 @@ const styles = StyleSheet.create({
 });
 
 const PaymentMethods = () => {
+
   const { start, paymentMethods } = useAdyenCheckout();
   const isDarkMode = useColorScheme() === 'dark';
   const contentBackgroundStyle = {
     backgroundColor: isDarkMode ? Colors.black : Colors.white,
   };
   const platformSpecificPayment =
     Platform.OS === 'ios' ? 'Apple Pay' : 'Google Pay';
+  const platformSpecificType =
+    Platform.OS === 'ios' ? 'applepay' : 'googlepay';
+  const isAvailable = (type) => { 
+    if (!paymentMethods) { return false; }
+    return paymentMethods.paymentMethods.find(x => x.type === type.toLowerCase()) != null
+  };
```

Comment:
Better practice is to wrap functions in `useCallback` hook, so they don't get recreated during the re-rendering step.

Also, you can use `Array.prototype.some` to verify if the payment method is in the array, so you don't have to do the check `!== null` at the end of the comparison statement.

```js
const isAvailable = useCallback(type => { 
   return paymentMethods.paymentMethods.length > 0 && paymentMethods.paymentMethods.some(pm => pm.type === type.toLowerCase());
}, [paymentMethods, paymentMethods.paymentMethods]);
```

Side note: It is a bit weird though to have `paymentMethods` object which has another `paymentMethods` property inside it 🤔

Reply:
> It is a bit weird though to have paymentMethods object which has another paymentMethods property inside it 🤔

Good point. I'll rename it to `paymentMethodsResponse`.